components/sup/src/manager/service/spec.rs

```diff
@@ -267,6 +264,50 @@ impl ServiceSpec {
         Ok(self)
     }
 
+    pub fn merge_svc_update(&mut self, svc_update: habitat_sup_protocol::ctl::SvcUpdate) {
+        if let Some(group) = svc_update.group {
+            self.group = group;
+        }
+        if let Some(bldr_url) = svc_update.bldr_url {
+            self.bldr_url = bldr_url;
+        }
+        if let Some(channel) = svc_update.bldr_channel {
+            self.channel = channel.into();
+        }
+        if let Some(topology) = svc_update.topology {
+            if let Some(topology) = Topology::from_i32(topology) {
```

Comment:
:thinking: Throwing an error might be better. If we'd gotten this far and can't parse a value, that would indicate that something went wrong (though it could indicate that a mismatch between otherwise-compatible client and server versions).

Properly done, it feels like it would be better to incorporate this a bit lower in the stack. That is, I'd like to raise the error at the point where we pull the protobuf off the wire. I think that would also require the creation of a separate-but-similar type. On one hand, that seems a bit awkward. On the other hand, it makes some amount of sense. The protobuf defines the communication messages between the client and the control gateway, whereas _this_ interaction is between the control gateway and the Supervisor internals, which is arguably a different interaction domain entirely. It felt a bit weird to be dealing with a protobuf type down here.

Carrying that to its logical extension could be a bit of work, though :frowning: 

Perhaps `error!` logging would be a sensible interim solution :sweat_smile: